src/main/java/uk/nhs/hee/tis/trainee/forms/model/AbstractForm.java

```diff
@@ -37,7 +37,9 @@ public abstract class AbstractForm {
   @Indexed
   @Field(value = "traineeTisId")
   private String traineeTisId;
+  private Boolean isArcp;
 
+  private UUID programmeMembershipId;
```

Comment:
This class will be used for other forms, like LTFT, in the future which are not related to PMs or ARCPs.
You could create an intermediate `AbstractLinkedForm` between this abstract and and the two Form R concrete classes.

Reply:
I think this comment is still valid, but given the different combinations of ARCP and which objects we might link forms to (I don't know whether we might need to link to placements in the future too) we probably need to keep these fields on the Form R classes instead of an abstract.

---

Using some interfaces might keep things more sensible, but it wouldn't change the data structure and the fields would still need to be declared on the FormRPartA/FormRPartB classes. So this could be something we do as part of the the LTFT work;

e.g.
```java
public interface ProgrammeMembershipLinked {
  public UUID getProgrammeMembershipId();
  public void setProgrammeMembershipId(UUID programmeMembershipId);
}

public interface ArcpLinked {
  public boolean getIsArcp();
  public void setIsArcp(boolean isArcp);
}

public class FormRPartA extends AbstractForm implements ProgrammeMembershipLinkedForm, ValidForArcpForm  {
  // Snip
  
  private UUID programmeMembershipId;
  private boolean isArcp;
   
  // Snip
}

public class FormRPartB extends AbstractForm implements ProgrammeMembershipLinkedForm, ValidForArcpForm  {
  // Snip
  
  private UUID programmeMembershipId;
  private boolean isArcp;
   
  // Snip
}

public class LessThanFullTime extends AbstractForm implements ProgrammeMembershipLinkedForm {
  // Snip
  
  private UUID programmeMembershipId;
   
  // Snip
}

```